sim/aviron/src/lib/Cpu.zig

```diff
@@ -141,6 +198,12 @@ pub fn run(cpu: *Cpu, mileage: ?u64) RunError!RunResult {
                     }
                 },
             }
+
+            // Check if the program requested exit via I/O
+            if (cpu.io.check_exit()) |exit_code| {
```

Comment:
This is a big janky: IO has to store whether we need to exit, so that the CPU knows to stop running.

I do this so that we don't just kill the process when the inferior writes to 0. This way we can do cleanup, dump debug info, etc. Before this made debugging annoying since a proper exit would just skip all my debug prints.